api/preprints/serializers.py

```diff
@@ -324,7 +323,7 @@ def update(self, preprint, validated_data):
                 detail='Cannot provide conflict of interest statement when has_coi is set to False.',
             )
 
-        if updated_has_data_links != 'no' and updated_why_no_data:
+        if updated_has_data_links != 'no':
```

Comment:
I managed to reproduce this problem locally by manually changing the database record, but the problem still exists, I believe this is the right solution.
```suggestion
        if updated_has_data_links != 'no' and updated_why_no_data and validated_data.get('data_links'):
```

Reply:
Removed these changes